src/Charcoal/Queue/AbstractQueueManager.php

```diff
@@ -456,15 +494,32 @@ public function totalQueuedItems()
      */
     public function totalChunks()
     {
-        return (int)ceil($this->totalQueuedItems() / $this->chunkSize());
+        $total = $this->totalQueuedItems();
+
+        $limit = $this->limit();
+        if ($limit > 0 && $total > $limit) {
```

Comment:
FYI: This portion should be okay but there could be edge cases to mixing `chunkSize` and `limit`.